ui/pages/settings/flask/snaps-list-tab/snap-list-tab.container.js

```diff
@@ -0,0 +1,64 @@
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+import { ENVIRONMENT_TYPE_POPUP } from '../../../../../shared/constants/app';
+import { getEnvironmentType } from '../../../../../app/scripts/lib/util';
+
+import {
+  SNAPS_VIEW_ROUTE,
+  SNAPS_LIST_ROUTE,
+} from '../../../../helpers/constants/routes';
+import { disableSnap, enableSnap } from '../../../../store/actions';
+import SnapListTab from './snap-list-tab.component';
+
+const mapStateToProps = (state, ownProps) => {
+  const { location, history } = ownProps;
+  const { pathname } = location;
+
+  const viewingSnap = Boolean(pathname.match(SNAPS_VIEW_ROUTE));
+  const pathNameTail = pathname.match(/[^/]+$/u)[0];
+  const envIsPopup = getEnvironmentType() === ENVIRONMENT_TYPE_POPUP;
+  const onClick = (_, snap) => {
+    const route = `${SNAPS_VIEW_ROUTE}/${window.btoa(
+      unescape(encodeURIComponent(snap.id)),
```

Comment:
Huh. What is going on here? I have never seen `unescape` used before, and apparently it's deprecated.

Why can't we just use `encodeURIComponent(snap.id)` here?

Reply:
Hmm not sure, @shanejonas -- is there a specific reason for the use of unescape?

Reply:
needs escape to handle all cases properly: https://developer.mozilla.org/en-US/docs/Glossary/Base64#the_unicode_problem

Reply:
Yeah but why base64 encode it in the first place? `encodeURIComponent` is all we need to make it URL-safe.